go/vm/lfvm/interpreter.go

```diff
@@ -793,7 +797,8 @@ func isWriteInstruction(opCode OpCode) bool {
 		1<<(LOG4-SSTORE) |
 		1<<(CREATE-SSTORE) |
 		1<<(CREATE2-SSTORE) |
-		1<<(SELFDESTRUCT-SSTORE)
+		1<<(SELFDESTRUCT-SSTORE) |
+		1<<(TSTORE-SSTORE)
```

Comment:
this is obscure.... can you explain to me what this does?

Reply:
This also took me a few minutes to understand, it is used to check whether a opcodes writes and is therefore not allowed in static mode. 
It uses the fact that SSTORE is the first opcode that is not allowed in static mode. For each operation listed the bit mask is set using the difference to SSTORE as the offset. In the line of the return it is then checked whether the opcode is smaller than SSTORE and the bit mask is set at the given offset.
I am pretty sure this is a go work around for better performance, but I am not sure.

Reply:
yes, this looks like a pretty brainy optimization. Funny that there is a benchmark, but no unit test ;)

Reply:
Jup, I remember building this with Jan when all of this was just a early prototype. This bit-masking is way faster than a lookup in a map or slice.

Anyway: I think `TSTORE` should not be in this list. TSTORE has no lasting effect, and is thus not a `WriteInstruction` that should fail in a static context. Running TSTOREs in static contexts is fine (at least in my understanding, and evm.codes also does not list a error condition in case of being used in a read-only context).

Reply:
I responded on the evmzero PR with links to the geth implementation and execution spec implementation.